domain/timeline/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/timeline/TimelineEvent.kt

```diff
@@ -42,25 +42,33 @@ data class TimelineEvent(
    * The date and time when the event occurred.
    */
   val timestamp: Instant,
+  /**
+   * An optional correlationId for this and any other TimelineEvents that occurred at the same time (i.e. within same
+   * atomic action). For example, this could be an update to a Goal and one or more of its child Steps.
+   */
+  val correlationId: UUID?,
 ) {
   companion object {
     fun newTimelineEvent(
+      reference: UUID = UUID.randomUUID(),
```

Comment:
Was this a deliberate change, or was it because you felt it was missing from the method args?
By allowing it to be passed as a method arg it means that the (production) code that is new'ing up a new timeline event could in theory specify it's own reference? 🤔 
Is that what we meant?

Reply:
I'm trying to recall why I did this! I think it was purely for consistency/symmetry, so that both parameters had `: UUID = UUID.randomUUID()` (i.e. my OCD got the better of me!). Happy to change it back, but I trust the clients in this case not to provide a value unless they need to.